File: Java/Array/Move all negative elements to end.java
//https://www.geeksforgeeks.org/problems/move-all-negative-elements-to-end1813/1
class Solution {
    public void segregateElements(int[] arr) {
        // Your code goes here
        List<Integer> pos = new ArrayList<>();
    List<Integer> neg = new ArrayList<>();

    // Separate positive and negative elements
    for (int num : arr) {
        if (num < 0) {
            neg.add(num);
        } else {
            pos.add(num);
        }
    }

    // Merge them back into the original array
    int j = 0;
    for (int num : pos) {
        arr[j++] = num;
    }
    for (int num : neg) {
        arr[j++] = num;
    }
    }
}

File: Python/Array/Move all negative elements to end.py
//https://www.geeksforgeeks.org/problems/move-all-negative-elements-to-end1813/1
class Solution:
    def segregateElements(self, arr):
        # Your code goes here
        pos = []
        neg = []
        for num in arr:
            if num >= 0:
                pos.append(num)
            else:
                neg.append(num)
        
        # Combine them back into the original array
        for i in range(len(pos)):
            arr[i] = pos[i]
        for i in range(len(neg)):
            arr[len(pos) + i] = neg[i]
